bundles/binding/org.openhab.binding.cardio2e/src/main/java/org/openhab/binding/cardio2e/internal/code/Cardio2eCurtainTransaction.java

```diff
@@ -0,0 +1,195 @@
+		return (openingPercentage == CARDIO2E_MIN_CURTAIN_OPENING_PERCENTAGE);
+	}
+
+	public void setCurtainClosed(boolean curtainClosed) {
+		setCurtainOpen(!curtainClosed);
+	}
+
+	public boolean getLastMoveWasUp() {
+		return lastMoveWasUp;
+	}
+
+	public boolean getLastMoveWasDown() {
+		return lastMoveWasDown;
+	}
+
+	public boolean isDataVerified() { //Returns true if object data matches with Cardio2e data specs and data is congruent.
+		boolean verified = false;
+		if (super.isDataVerified()) {
+			switch (getTransactionType()) {
+			case ACK:
+			case GET:
+				verified = (getObjectNumber() != -1);
+				break;
+			case NACK:
+				verified = (getErrorCode() != -1);
+				break;
+			case INFORMATION:
+			case SET:
+				verified = ((getObjectNumber() != -1) && (openingPercentage != -1));
+				break;
+			}
+		}
+		return verified;
+	}
+	
+	public String toString () {
+		String returnString = null;
+		if (isDataVerified()) {
+			switch (getTransactionType()) {
+			case ACK:
+			case GET:
+				returnString = String.format("%s%s %s %d%s",CARDIO2E_START_TRANSACTION_INITIATOR,getTransactionType().symbol,getObjectType().symbol,getObjectNumber(),CARDIO2E_END_TRANSACTION_CHARACTER);
+				break;
+			case NACK:
+				if (getObjectNumber() == -1) {
+					returnString = String.format("%s%s %s %d%s",CARDIO2E_START_TRANSACTION_INITIATOR,getTransactionType().symbol,getObjectType().symbol,getErrorCode(),CARDIO2E_END_TRANSACTION_CHARACTER);
+				}
+				else {
+					returnString = String.format("%s%s %s %d %d%s",CARDIO2E_START_TRANSACTION_INITIATOR,getTransactionType().symbol,getObjectType().symbol,getObjectNumber(),getErrorCode(),CARDIO2E_END_TRANSACTION_CHARACTER);
+				}
+				break;
+			case INFORMATION:
+			case SET:
+				returnString = String.format("%s%s %s %d %d%s",CARDIO2E_START_TRANSACTION_INITIATOR,getTransactionType().symbol,getObjectType().symbol,getObjectNumber(),openingPercentage,CARDIO2E_END_TRANSACTION_CHARACTER);
+				break;
+			}			
+		}
+		return returnString;
+	}
+
+	public static void setSmartSendingEnabledClass(boolean enabled) {
+		Cardio2eCurtainTransaction.smartSendingEnabledClass = enabled;
+	}
+
+	public static boolean getSmartSendingEnabledClass() {
+		//This function replaces superclass Cardio2eTransaction getSmartSendingEnabledClass() function.
+		return Cardio2eCurtainTransaction.smartSendingEnabledClass;
+	}
+
+	public boolean isSmartSendingEnabled() { //Returns true if this transaction or its class is smart sending enabled.
+		//This function replaces superclass Cardio2eTransaction isSmartSendingEnabled() function.
+		return (Cardio2eCurtainTransaction.getSmartSendingEnabledClass() || smartSendingEnabledTransaction);	
+	}
+	
+	public boolean smartSendingCanReplaceLikeMe() {
+		//This function has been replaced in this Cardio2eTransaction subclass in order to return true.
+		//When returns true, smart sending will can delete the transactions like this from the send buffer and add a new transaction that replaces them.
```

Comment:
It seems my comment was confusing. I have not been able to translate it well into English. I meant "will be able to delete"